pkg/morserino_com/list_com_test.go

```diff
@@ -0,0 +1,299 @@
+package morserino_com
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.bug.st/serial/enumerator"
+)
+
+//
+// ============ Mock definitions ==================
+
+//Happy case mock
+type mockEnumeratePorts_OK struct{}
+
+//Happy case mocked method
+func (e mockEnumeratePorts_OK) GetDetailedPortsList() ([]*enumerator.PortDetails, error) {
+	happyCaseList := []*enumerator.PortDetails{}
+
+	item1 := enumerator.PortDetails{
+		Name:         "COM Port Name",
+		IsUSB:        true,
+		VID:          "XXXX",
+		PID:          "YYYY",
+		SerialNumber: "001",
+	}
+	item2 := enumerator.PortDetails{
+		Name:         "Morserino Port",
+		IsUSB:        true,
+		VID:          "10C4",
+		PID:          "EA60",
+		SerialNumber: "002",
+	}
+	happyCaseList = append(happyCaseList, &item1)
+	happyCaseList = append(happyCaseList, &item2)
+
+	return happyCaseList, nil
+}
+
+// Mock that returns an error
+type mockEnumeratePorts_error struct{}
+
+// Mocked method that returns an error
+func (e mockEnumeratePorts_error) GetDetailedPortsList() ([]*enumerator.PortDetails, error) {
+	return nil, fmt.Errorf("An error occured")
+}
+
+// Mock that found no ports
+type mockEnumeratePorts_noPort struct{}
+
+// Mocked method that found no ports
+func (e mockEnumeratePorts_noPort) GetDetailedPortsList() ([]*enumerator.PortDetails, error) {
+	emptyList := []*enumerator.PortDetails{}
+
+	return emptyList, nil
+}
+
+//
+// =================  Tests ====================
+//
+
+func Test_Get_com_list_HappyCase(t *testing.T) {
+
+	//Happy case mock
+	var mockEnumPorts mockEnumeratePorts_OK
+
+	// System under test
+	comList, err := Get_com_list(mockEnumPorts)
+
+	//Expected Port List
+	targetPortList := []comPortItem{
+		{
+			portName:        "COM Port Name",
+			usbVendorID:     "XXXX",
+			usbProductID:    "YYYY",
+			serialNumber:    "001",
+			isMorserinoPort: false,
+		},
+		{
+			portName:        "Morserino Port",
+			usbVendorID:     "10C4",
+			usbProductID:    "EA60",
+			serialNumber:    "002",
+			isMorserinoPort: true,
+		},
+	}
+
+	//validate results
+	if assert.NoError(t, err) {
+		assert.Equal(t, comList.nbrOfPorts, 2)
+		assert.Equal(t, comList.nbrOfMorserinoPorts, 1)
+		assert.Equal(t, comList.morserinoPortName, "Morserino Port")
+		assert.Equal(t, comList.portList, targetPortList)
+	}
```

Comment:
```suggestion
	require.NoError(t, err)
	
	assert.Equal(t, comList.nbrOfPorts, 2)
	assert.Equal(t, comList.nbrOfMorserinoPorts, 1)
	assert.Equal(t, comList.morserinoPortName, "Morserino Port")
	assert.Equal(t, comList.portList, targetPortList)
```

You can use `github.com/stretchr/testify/require` to stop the test execution if your assertion fails. It saves you  an indentation level here.

Reply:
Cool. Learned something new